### PR TITLE
Feat: fetch tricks from firestore

### PR DESCRIPTION
**Pull request by berk245**
## Description

Setup Firestore collections and access them on page load. 

Current status:
- We have a users collection, in which every user has a document. (If a user doesn't have a db entry, we automatically create a new one on login)
- In this user document, we have a sub collection called tricks. 
- Currently we only have test data connected to test@test.com account. Until we implement CRUD functions for tricks, you can manually add tricks in Firebase console. I am happy to help if that's needed. In any case, let's do our best to add the CRUD functions soon

## Task

[Notion Task Link ](https://www.notion.so/d077118807034bbfb517f36c20d2a304?v=345833d9689145a2b5a197646677c93e&p=e9863ab223344777b724147fb8461ce8&pm=s)

## Screenshots (optional)
image

### CODE REVIEW

**Review by tonyjojo — APPROVED** [review]
Beautiful, love to see the gallery populating with actual data

File: src/lib/store/store.js [hunk 1]
```diff
@@ -2,5 +2,6 @@ import { writable } from 'svelte/store';
 
 export const authStore = writable({
 	user: null,
-	loading: true
+	loading: true,
+	tricks: []
```
Comment (niccoloMG) [inline src/lib/store/store.js:6]: I think we should either change the store name or create separate stores. wdyt?

File: src/lib/handlers/db.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { db } from '$lib/firebase/firebase';
+import { collection, doc, getDoc, getDocs, setDoc } from 'firebase/firestore';
+
+export const getUserTricks = async (user) => {
+	if (!user) return [];
+	const userRef = doc(db, 'users', user.uid);
+	const userSnap = await getDoc(userRef);
+	if (!userSnap.exists()) {
+		console.log('No data for this user found in the collection.');
+		const newUser = doc(db, 'users', user.uid);
```
Comment (niccoloMG) [inline src/lib/handlers/db.js]: not a big deal but we can extract the creating user document function to make it more readable

File: src/lib/store/store.js (hunk 1 above)
Comment (berk245) [reply src/lib/store/store.js:6]: Good point! let's create a ticket for rethinking & improving the store structure

File: src/lib/handlers/db.js (hunk 2 above)
Comment (berk245) [reply src/lib/handlers/db.js]: Great point! created a separate function with the last commit

**Pull request merged**